src/commands/dev/socket.rs

```diff
@@ -115,44 +115,40 @@ async fn print_ws_messages(
     mut read: SplitStream<WebSocketStream<MaybeTlsStream<TcpStream>>>,
 ) -> Result<()> {
     while let Some(message) = read.next().await {
-        match message {
-            Ok(message) => {
-                let message_text = message.into_text().unwrap();
-                log::info!("{}", &message_text);
-
-                let parsed_message: Result<protocol::Runtime> = serde_json::from_str(&message_text)
-                    .map_err(|e| anyhow!("Failed to parse event:\n{}", e));
-
-                match parsed_message {
-                    Ok(protocol::Runtime::Event(ExceptionThrown(params))) => {
-                        let default_description = "N/A".to_string();
-                        let description = params
-                            .exception_details
-                            .exception
-                            .description
-                            .as_ref()
-                            .unwrap_or(&default_description);
-
-                        StdOut::message(&format!(
-                            "{} at line {:?}, col {:?}",
-                            description,
-                            params.exception_details.line_number,
-                            params.exception_details.column_number,
-                        ));
-
-                        let json_parse = serde_json::to_value(params.clone());
-                        print_json(json_parse, format!("{:?}", params));
-                    }
-                    Ok(protocol::Runtime::Event(event)) => {
-                        // Try to parse json to pretty print, otherwise just print string
-                        let json_parse: Result<serde_json::Value, serde_json::Error> =
-                            serde_json::from_str(&*event.to_string());
-                        print_json(json_parse, event.to_string());
-                    }
-                    _ => {}
-                }
+        let message = message?;
+        let message_text = message.into_text().unwrap();
+        log::info!("{}", &message_text);
+
+        let parsed_message: Result<protocol::Runtime> = serde_json::from_str(&message_text)
+            .map_err(|e| anyhow!("Failed to parse event:\n{}", e));
```

Comment:
I just noticed we never actually handle this error - it's silently discarded in the `match`. Should it use `?` instead?
```suggestion
        let parsed_message: Result<protocol::Runtime> = serde_json::from_str(&message_text)
            .map_err(|e| anyhow!("Failed to parse event:\n{}", e))?;
```

Reply:
I'm not sure if that would work, if the message failed to parse than it's likely a heartbeat message and therefore we just want to noop and move on. If we return the Err out of the function the `try_join!` on line 61 might kill everything but if you try and it and it doesn't stop dev after receiving heartbeat messages then I think it's fine.

Reply:
Ok, we should probably at least log it then. Right now the `anyhow!` doesn't do anything, it just allocates and then immediately deallocates.

Reply:
Yeah that might be a good idea, we have that `log::info` on line 121 but maybe a debug for the result if it failed to parse

Reply:
Great, added that :)